### PR TITLE
Feature/eng 11496 eng 11494 update user info

### PR DESCRIPTION
**Pull request by vzotov**
@Glympse/web PTAL

- Added `core.setUserName` and `core.setUserAvatar` endpoints
  - setUserName(name: string)
  - setUserAvatar(urlOrArrayBuffer: string|arraybuffer)
- Added `core.getUserInfo` endpoint
- new events
  - `UserNameUpdateStatus` - returns user info
  - `UserAvatarUpdateStatus` - returns avatar url
  - `UserInfo` - returns user's data

### CODE REVIEW

File: app/src/adapter/models/Account.js [hunk 1]
```diff
@@ -110,6 +110,120 @@ define(function(require, exports, module)
...
+					{
+						result.status = true;
+					}
+					result.response = data.response;
+				}
+				controller.notify(Account.UserAvatarUpdateStatus, result);
+			}
+		};
+
+		this.getUserInfo = function ()
```
Comment (j5kay) [inline app/src/adapter/models/Account.js]: This should accept an optional string-based user-id param. If null/undefined, use "self". Update in other places (CoreController, README.md, etc.) accordingly

File: app/src/adapter/models/Account.js [hunk 2]
```diff
@@ -261,8 +375,14 @@ define(function(require, exports, module)
 	}
 
 	// Account defines
+	// Events
 	Account.InitComplete = 'AccountInitComplete';
 	Account.CreateStatus = 'AccountCreateStatus';
+	Account.UserInfo = 'UserInfo';
```
Comment (j5kay) [inline app/src/adapter/models/Account.js]: Needs to be 'UserInfoStatus' to stay consistent with other status event names

File: app/src/adapter/CoreController.js [hunk 3]
```diff
@@ -32,21 +32,22 @@ define(function(require, exports, module)
...
-					break;
-				}
-
-				default:
-				{
-					dbg('Unknown msg: "' + msg + '"', args);
-					break;
-				}
-			}
+			controller.notify(msg, args);
```
Comment (j5kay) [inline app/src/adapter/CoreController.js]: While simpler, the emphasis here should look at caching semantics and/or additional processing to simplify the data before it's is bubbled up to the main controller. Caching will help to reduce API call patterns and some errors/data formatting should be handled internally.

Should be fine for now, and I suspect you left the code commented for these scenarios.

File: app/src/adapter/models/Account.js [hunk 4]
```diff
@@ -110,6 +110,120 @@ define(function(require, exports, module)
 			}
 		};
 
+		this.setName = function (newName)
```
Comment (j5kay) [inline app/src/adapter/models/Account.js]: Did you run `grunt` before submitting? :)

**Comment by vzotov** [comment]
OK, done

File: README.md [hunk 5]
```diff
@@ -640,7 +640,8 @@ The following APIs are only available to consumers of the GA when running in
 client-mode. These are also specifed in `GlympseAdapterDefines.CORE.REQUESTS_LOCAL`:
 
 - `accountCreate()`: Creates account, sends event `AccountCreateStatus` as a result.
-- `getUserInfo()`: Gets user data, sends event `UserInfo` with the result
+- `getUserInfo(userId: string)`: Gets user data by `userId`, if it is not defined or empty
```
Comment (j5kay) [inline README.md]: Please change to description to:

Gets info for the current logged in user (name, avatar URL, etc.). If the optional `userId` param is non-null, info is retrieved for the specified user instead.

**Pull request merged**